src/main/java/com/epam/aidial/core/util/ProxyUtil.java

```diff
@@ -83,7 +87,7 @@ private static int contentLength(MultiMap header, int defaultValue) {
         return defaultValue;
     }
 
-    public static void collectAttachedFiles(ObjectNode tree, ApiKeyData apiKeyData) {
+    public static void collectAttachedFiles(ObjectNode tree, ApiKeyData apiKeyData, Function<String, Boolean> checkAccessFn) throws Exception {
```

Comment:
Consumer<String> accessChecker?

Reply:
Do you want to throw an exception in the consumer auction directly?

Reply:
Yes

Reply:
agreed to keep it as is